aiidalab_launch/util.py

```diff
@@ -127,7 +127,7 @@ def iter_to_queue() -> None:
     return yield_queue_items()
 
 
-def get_latest_version(timeout: float = 0.1) -> Optional[Version]:
+def get_latest_version(timeout: float = 0.1) -> Union[LegacyVersion, Version, None]:
```

Comment:
This is an unrelated mypy fix

Reply:
Could you elaborate a bit further on this? Does this occur with a specific mypy version? We are running [mypy checks as part of the CI](https://github.com/aiidalab/aiidalab-launch/blob/73fe854e525d1c0adfa1f92b1aa97842df5a5c16/.pre-commit-config.yaml#L37-L45) so that should have come up.

Reply:
Here's the output from conda list

<details>

```
_libgcc_mutex             0.1                        main  
_openmp_mutex             5.1                       1_gnu  
aiidalab-launch           2022.1013                 dev_0    <develop>
appdirs                   1.4.4                    pypi_0    pypi
attrs                     21.4.0                   pypi_0    pypi
bumpver                   2021.1114                pypi_0    pypi
bzip2                     1.0.8                h7f98852_4    conda-forge
ca-certificates           2022.6.15            ha878542_0    conda-forge
cattrs                    1.10.0                   pypi_0    pypi
certifi                   2022.6.15        py39hf3d152e_0    conda-forge
cffi                      1.15.0           py39h4bc2ebd_0    conda-forge
cfgv                      3.3.1              pyhd8ed1ab_0    conda-forge
charset-normalizer        2.0.12                   pypi_0    pypi
click                     8.0.3                    pypi_0    pypi
click-spinner             0.1.10                   pypi_0    pypi
colorama                  0.4.4                    pypi_0    pypi
coverage                  6.4.1                    pypi_0    pypi
distlib                   0.3.4              pyhd8ed1ab_0    conda-forge
docker                    5.0.3                    pypi_0    pypi
dunamai                   1.7.0                    pypi_0    pypi
filelock                  3.7.1              pyhd8ed1ab_0    conda-forge
identify                  2.5.1              pyhd8ed1ab_0    conda-forge
idna                      3.3                      pypi_0    pypi
iniconfig                 1.1.1                    pypi_0    pypi
ld_impl_linux-64          2.38                 h1181459_1  
lexid                     2021.1006                pypi_0    pypi
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 11.2.0               h1234567_1  
libgomp                   11.2.0               h1234567_1  
libnsl                    2.0.0                h7f98852_0    conda-forge
libstdcxx-ng              11.2.0               h1234567_1  
libuuid                   2.32.1            h7f98852_1000    conda-forge
libzlib                   1.2.11            h166bdaf_1014    conda-forge
mypy                      0.910              pyhd3eb1b0_0  
mypy_extensions           0.4.3            py39h06a4308_1  
ncurses                   6.3                  h7f8727e_2  
nodeenv                   1.6.0              pyhd8ed1ab_0    conda-forge
openssl                   3.0.3                h166bdaf_0    conda-forge
packaging                 21.3                     pypi_0    pypi
pathlib2                  2.3.7.post1              pypi_0    pypi
pip                       21.2.4           py39h06a4308_0  
platformdirs              2.5.1              pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0                    pypi_0    pypi
pre-commit                2.19.0           py39hf3d152e_0    conda-forge
psutil                    5.8.0            py39h27cfd23_1  
py                        1.11.0                   pypi_0    pypi
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.9                    pypi_0    pypi
pytest                    6.2.5                    pypi_0    pypi
pytest-asyncio            0.17.2                   pypi_0    pypi
pytest-cov                3.0.0                    pypi_0    pypi
python                    3.9.12          h2660328_1_cpython    conda-forge
python-dotenv             0.19.1                   pypi_0    pypi
python_abi                3.9                      2_cp39    conda-forge
pyyaml                    6.0              py39hb9d737c_4    conda-forge
readline                  8.1.2                h7f8727e_1  
requests                  2.26.0                   pypi_0    pypi
requests-cache            0.9.1                    pypi_0    pypi
responses                 0.18.0                   pypi_0    pypi
setuptools                61.2.0           py39h06a4308_0  
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.38.5               h4ff8645_0    conda-forge
tabulate                  0.8.9                    pypi_0    pypi
tk                        8.6.12               h27826a3_0    conda-forge
toml                      0.10.2             pyhd3eb1b0_0  
tomli                     2.0.1                    pypi_0    pypi
types-click-spinner       0.1.11                   pypi_0    pypi
types-pyyaml              6.0.8                    pypi_0    pypi
types-redis               4.2.7                    pypi_0    pypi
types-requests            2.27.30                  pypi_0    pypi
types-tabulate            0.8.9                    pypi_0    pypi
types-toml                0.10.7                   pypi_0    pypi
types-ujson               5.3.0                    pypi_0    pypi
types-urllib3             1.26.15                  pypi_0    pypi
typing_extensions         4.1.1              pyh06a4308_0  
tzdata                    2022a                hda174b7_0  
ukkonen                   1.0.1            py39hf939315_2    conda-forge
url-normalize             1.4.3                    pypi_0    pypi
urllib3                   1.26.9                   pypi_0    pypi
virtualenv                20.14.1          py39hf3d152e_0    conda-forge
websocket-client          1.3.2                    pypi_0    pypi
wheel                     0.37.1             pyhd3eb1b0_0  
xz                        5.2.5                h7f8727e_1  
yaml                      0.2.5                h7f98852_2    conda-forge
zlib                      1.2.11            h166bdaf_1014    conda-forge
```
</details>

Reply:
We are using mypy version 0.961, not 0.910 . I am doubtful that this is the issue, but would be at least worth to check. Are the pre-commit hooks passing for you?

Reply:
Pre-commit hooks are passing. I have updated to 0.961 but the issue persisted. But I am happy to revert this at the end (keeping for now so mypy output is clean during development).